File: api/routes/visuals.js
const fs = require('fs')
const express = require('express')
const router = express.Router()


// bar chart
router.get('/bar', (req, res) => {
    data = fs.readFileSync('./public/data/random-categorical.json')
    res.json(JSON.parse(data))
})

// scatter plot
router.get('/scatter', (req, res) => {
    let data = []

    for (let i = 0; i < 100; i++) {
        let x, y, radius, color, opacity
        x = Math.random() * 500
        y = Math.random() * 500
        radius = Math.random() * (20 - 5) + 5

        color = Math.random()

        if (color < .66) {
            if (color < .33) {
                color = 'steelblue'
            } else {
                color = 'orange'
            }
        } else {
            color = 'firebrick'
        }

        opacity = Math.random() * (.75 - .1) + .1

        data.push({ x, y, radius, color, opacity })
    }

    res.json(data)
})

module.exports = router

File: api/routes/visuals.py
import random

from bson import json_util
from flask import Blueprint, jsonify

from api.models.visuals import categorical_dummy

bp = Blueprint('visuals', __name__)

# @visuals.route('/bar')
# def bar_chart():
#     data = categorical_dummy.find()
#     return json_util.dumps(data)

@bp.route('/scatter')
def scatter_plot():
    data = [{
        'x': random.random() * 500,
        'y': random.random() * 500,
        'radius': random.random() * (20 - 5) + 5,
        'color': random.choice(['steelblue', 'orange', 'firebrick'])
    } for i in range(100)]

    return jsonify(data)

# @visuals.route('/gapminder')
# def gapminder_viz():
#     data = gapminder.find({'continent': 'asia', 'year': {'$gt': 2000}}, {'_id': 0})
#     return json_util.dumps(data)
